keps/sig-release/1731-publishing-packages/README.md

```diff
@@ -207,20 +398,60 @@ promotion is an optional part of this KEP. As an intermediate solution we can
-all packages for all architectures is fast enough to be done as part of the
-release process. If it turns out this step takes too long, we need to think
-about doing the package building & publishing asynchronous to the release
-process (see also: [Risks](#risks-and-mitigations)).
+At the time of writing this KEP, there are no Go libraries for working with OBS that we could use to integrate directly
+with `krel`. Eventually, we could evaluate if it makes sense to build such a library for our purposes. Until then,
+we'll use `osc` directly (by exec-ing), which also requires adding `osc` to our build images.
+
+[obs-rabbitmq]: https://rabbit.opensuse.org/
+
+#### Authentication to OBS and User Management
+
+The concept of API tokens in OBS is very limited and provides access only to a very few endpoints. In other words,
+it's not possible to use API tokens for publishing to OBS. Instead, we need to create some sort of a service account
+to be used when publishing packages. This is one time operation that can be done by SIG Release Leads.
+
+Users are managed manually via the OBS web interface. SIG Release Leads must have access to add/remove users from
+our OBS project. Release Managers should be given read/write access, so they can maintain and create projects and
+packages.
+
+#### How are packages used?
+
+The End Users can configure their systems’ package managers to use
+those different `${channel}`s of a kubernetes `${k8s_release}` for their
+corresponding package manager.
+
+A configuration for the package managers might look something like:
+
+- deb (deb822 formatted version):
+  ```
+  Types: deb
+  URIs: http://packages.kubernetes.io/core:/stable:/v1.26/deb/
+  Suites: ./
+  Signed-By: /etc/keyrings/kubernetes-keyring.gpg
+  ```
+- rpm/yum:
+  ```
+  [kubernetes]
+  name=Kubernetes
+  type=rpm-md
+  baseurl=https://packages.kubernetes.io/core:/stable:/v1.26:/build/rpm/
+  gpgcheck=1
+  gpgkey=https://packages.kubernetes.io/core:/stable:/v1.26:/build/rpm/repodata/repomd.xml.key
+  enabled=1
+  ```
+
+**Note**: The (base) URL for packages will be changed from `{apt,yum}.kubernetes.io` to `packages.kubernetes.io`.
+This is to better support the OBS workflow. This is consider as a kind of breaking change, however, since there
+are other manual migration steps needed (e.g. changing the GPG key), we don't consider this as a problem.
+
+Different architectures will be published into the same repos, it is up to the package managers to pull and install the correct package for the target platform.
 
 ### Risks and Mitigations
 
-- _Risk_: We don't find a proper way to share secrets like the signing key
-  _Mitigation_: Using a third party tool like 1Password
+- _Risk_: The OBS installation provided by openSUSE is unable to serve the load generated by the Kubernetes project
+  _Mitigation_: We can host our own mirrors and take some load from openSUSE (e.g. on Equinix Metal)
```

Comment:
Should we aim to do anyway this for full ownership of the infrastructure?

Reply:
One mitigation is to serve to packages for object storage service. we have now access to AWS services. 
Can we sync packages from OBS to S3 ?

Reply:
> Should we aim to do anyway this for full ownership of the infrastructure?

Doing this is not giving us the full ownership of the infrastructure. The full ownership of the infrastructure would include hosting a whole OBS instance ourselves, which at this point we want to avoid because of:

* increased costs
* problems with managing the GPG key

> Only the OBS admins can set up a push sync hook (that is, it would sync on publish immediately).

We discussed that on the last OBS team sync meeting (back in November) and the OBS team is okay with doing that. We want to try without this for now, and add new mirrors only if needed.

Reply:
This will be documented as part of the KEP.